libs/cmdio/render.go

```diff
@@ -46,8 +50,126 @@ func Heredoc(tmpl string) (trimmed string) {
 	return strings.TrimSpace(trimmed)
 }
 
-func renderJson(w io.Writer, v any) error {
-	pretty, err := fancyJSON(v)
+// writeFlusher represents a buffered writer that can be flushed. This is useful when
+// buffering writing a large number of resources (such as during a list API).
+type writeFlusher interface {
+	io.Writer
+	Flush() error
+}
+
+type jsonRenderer interface {
+	// Render an object as JSON to the provided writeFlusher.
+	renderJson(context.Context, writeFlusher) error
+}
+
+type textRenderer interface {
+	// Render an object as text to the provided writeFlusher.
+	renderText(context.Context, writeFlusher) error
+}
+
+type templateRenderer interface {
+	// Render an object using the provided template and write to the provided tabwriter.Writer.
+	renderTemplate(context.Context, *template.Template, *tabwriter.Writer) error
+}
+
+type readerRenderer struct {
+	reader io.Reader
+}
+
+func (r readerRenderer) renderText(_ context.Context, w writeFlusher) error {
+	_, err := io.Copy(w, r.reader)
+	if err != nil {
+		return err
+	}
+	return w.Flush()
+}
+
+type iteratorRenderer struct {
+	t          reflectIterator
+	bufferSize int
+}
+
+func (ir iteratorRenderer) getBufferSize() int {
+	if ir.bufferSize == 0 {
+		return 100
```

Comment:
I tried this with the jobs command and we still need 4 API calls to fill this buffer. If I use the `--limit` flag (which we should surface differently, tbh), then I can make a single call return 100 items.

How could we make the first page show up faster than waiting for 4 calls (still takes a couple secs)?

Not blocking of course, but since this is changing anyway, we might as well make as fast as possible.

Reply:
I just lowered the buffer size to 20, which makes it appear much faster.